doc/source/ray-core/_examples/datasets_train/datasets_train.py

```diff
@@ -580,7 +580,7 @@ def train_func(config):
         read_dataset(data_path)
     )
 
-    num_columns = len(train_dataset.schema().names)
+    num_columns = len(train_dataset.schema(fetch_if_missing=True).names)
```

Comment:
Should we change the `fetch_if_missing` default value to `True` in `ds.schema()`? I.e. should we transparently trigger execution when fetching metadata on a lazy dataset?

Reply:
> should we transparently trigger execution when fetching metadata on a lazy dataset?

Yes agreed. Do we want to do in a separate PR ? https://github.com/ray-project/ray/pull/31286#issuecomment-1372976970 .

Reply:
Yep doing that in a separate PR sounds good!